src/Raven.Client/Util/WeakReferencingTimer.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Threading;
+
+namespace Raven.Client.Util;
+
+internal class WeakReferencingTimer : IDisposable
+{
+    internal delegate void WeakReferencingTimerCallback(object state);
+
+    private readonly Timer _timer;
+
+    public WeakReferencingTimer(WeakReferencingTimerCallback callback, object state, TimeSpan dueTime, TimeSpan period)
+    {
+        if (callback == null)
```

Comment:
What about the callback itself? That may hold a reference to the instance, no?

Reply:
You might see something that I'm missing here. In case of our `RequestExecutor` we pass the _non static_ method as the callback (so we reference to the instance this way):

```
public class RequestExecutor
{
    private WeakReferencingTimer _updateTopologyTimer;

    private void InitializeUpdateTopologyTimer()
    {
        _updateTopologyTimer = new WeakReferencingTimer(UpdateTopologyCallback, null, TimeSpan.FromMinutes(5), TimeSpan.FromMinutes(5));
    }
    
    internal async void UpdateTopologyCallback(object _)
    {
        var selector = _nodeSelector; // private member of request executor
        if (selector == null || selector.Topology == null)
            return;

        ...
    }
}
```

Such usage doesn't cause any issues. The finalizer of `RequestExecutor` is called.

Reply:
The problem was different - the reference to the callback could be GCed

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/17951/commits/9f1688df421103f6d2ed47c29a38a4c0b975b1dd